src/braket/aws/aws_session.py

```diff
@@ -687,12 +679,14 @@ def parse_s3_uri(s3_uri: str) -> tuple[str, str]:
         try:
             # Object URL e.g. https://my-bucket.s3.us-west-2.amazonaws.com/my/key
             # S3 URI e.g. s3://my-bucket/my/key
-            s3_uri_match = re.match("^https://([^./]+).[sS]3.[^/]+/(.*)$", s3_uri) or re.match(
-                "^[sS]3://([^./]+)/(.*)$", s3_uri
+            s3_uri_match = re.match("^https://([^./]+).[sS]3.[^/]+/(.+)$", s3_uri) or re.match(
```

Comment:
Why are we changing the logic as part of this PR?

Reply:
I can break this into its own PR

Reply:
Moved out into https://github.com/amazon-braket/amazon-braket-sdk-python/pull/860. I will rebase from main once the PR is merged.